packages/sdk-js/src/core.ts

```diff
@@ -58,6 +61,49 @@ function getForcedVariations(ctx: EvalContext) {
   }
 }
 
+async function safeCall(fn: () => void | Promise<void>) {
+  try {
+    await fn();
+  } catch (e) {
+    // Do nothing
+  }
+}
+
+function onExperimentViewed(
+  ctx: EvalContext,
+  experiment: Experiment<unknown>,
+  result: Result<unknown>
+) {
+  const calls: Promise<void>[] = [];
+
+  if (ctx.global.trackingCallback) {
+    const cb = ctx.global.trackingCallback;
+    calls.push(safeCall(() => cb(experiment, result, ctx.user)));
+  }
+  if (ctx.user.trackingCallback) {
+    const cb = ctx.user.trackingCallback;
+    calls.push(safeCall(() => cb(experiment, result)));
+  }
+  if (ctx.global.eventLogger) {
```

Comment:
can this potentially replace the trackingCallback? If so, do we need to do async event logging judo for redirect tests?

Also, track the user attributes, not just hash attribute?

Reply:
Yes, this could replace trackingCallback.  It's more opinionated about the shape of the data and also includes feature usage events.

This only includes properties specific to the event.  Within the event logger, you can call `gb.getAttributes()` to get all of the attributes to log.